docs/content/customization/aws/network.md

```diff
@@ -0,0 +1,65 @@
++++
+title = "Network"
++++
+
+The network customization allows the user to specify existing infrastructure to use for the cluster.
+
+This customization will be available when the
+[provider-specific cluster configuration patch]({{< ref "..">}}) is included in the `ClusterClass`.
+
+## Example
+
+To specify existing AWS VPC, use the following configuration:
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: <NAME>
+spec:
+  topology:
+    variables:
+      - name: clusterConfig
+        value:
+          aws:
+            network:
+              vpc:
+                id: vpc-1234567890
+```
+
+To also specify existing AWS Subnets, use the following configuration:
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: <NAME>
+spec:
+  topology:
+    variables:
+      - name: clusterConfig
+        value:
+          aws:
+            network:
+              vpc:
+                id: vpc-1234567890
```

Comment:
are we plan to support all the vpcSpec properties in future? https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/api/v1beta1/network_types.go#L184

Reply:
Yeah I can see us adding support for things like the CIDR